src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -54,7 +54,7 @@ public void GetServerUrl(AgentSettings agentSettings, CommandSettings command)
             agentSettings.ServerUrl = command.GetUrl();
         }
 
-        public async Task GetPoolId(AgentSettings agentSettings, CommandSettings command)
+        public virtual async Task GetPoolId(AgentSettings agentSettings, CommandSettings command)
```

Comment:
this method looks weird.
can you just do GetPoolIdAsync() inline, like what you had in the overwrite one.

Reply:
GetPoolIdAsync() is a private method only called by one method.

Reply:
I will have only one method GetPoolId(). will remove GetPoolIdAsync